File: drake_cmake_installed/src/find_resource/find_resource_example.cc
// SPDX-License-Identifier: MIT-0

///
/// @brief  A simple find_resource test for locating installed drake resources.
///

#include <iostream>
#include <stdexcept>

#include <drake/common/find_resource.h>

namespace drake_external_examples {
namespace {

int main() {
  drake::FindResourceOrThrow(
      "drake/manipulation/models/iiwa_description/urdf/"
      "iiwa14_primitive_collision.urdf");

  try {
    drake::FindResourceOrThrow("nobody_home.urdf");
    std::cerr << "Should have thrown" << std::endl;
    return 1;
  } catch (const std::runtime_error&) {
  }

  return 0;
}

}  // namespace
}  // namespace drake_external_examples

int main() { return drake_external_examples::main(); }

File: drake_cmake_external/drake_external_examples/src/find_resource/find_resource_example.py
# SPDX-License-Identifier: MIT-0

"""Test the Drake find resource API."""

import logging

from pydrake.common import FindResourceOrThrow

# If you have trouble finding resources, you can enable debug logging to see
# how `FindResource*` is searching.
logging.getLogger("drake").setLevel(logging.DEBUG)

FindResourceOrThrow("drake/examples/pendulum/Pendulum.urdf")
